src/main/java/Duke/task/Event.java

```diff
@@ -1,79 +1,75 @@
 package duke.task;
 
-import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
 import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
 
 /**
- * Represents an event task. An event task has a date and time that it will occur.
+ * Represents an event task. An event task has a start and end time.
  */
 public class Event extends Task {
 
-    private String at;
-    private LocalDate date;
-    private int startTime;
-    private int endTime;
+    private String timingStr;
+    private LocalDateTime timing;
+    private LocalTime endTime;
 
-    public Event(String details, String at, String date, int startTime, int endTime) {
+    public Event(String details, String timing) {
         super(details);
-        this.at = at;
-        this.date = date == null
-                ? null
-                : LocalDate.parse(date);
-        this.startTime = startTime;
-        this.endTime = endTime;
+        if (timing.length() == 20) {
+            try {
+                this.timing = LocalDateTime.parse(timing.substring(0, 15),
+                        DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
+                endTime = LocalTime.parse(timing.substring(16),
+                        DateTimeFormatter.ofPattern("HHmm"));
+            } catch (DateTimeParseException e) {
+                timingStr = timing;
+                this.timing = null;
+                this.endTime = null;
+            }
+        } else {
+            timingStr = timing;
+            this.timing = null;
+            this.endTime = null;
```

Comment:
Perhaps being consistent with using this to refer to class variables would be better?